scripts/release-notes.py

```diff
@@ -1,67 +1,133 @@
+
+    if not num_commits:
+        return
+
+    # Load commits
+    data = urllib.parse.urlencode({'per_page': num_commits})
+    req = Request(commits_url + '?' + data)
+    print(req.full_url)
+    req.add_header('Authorization', f'token {token}')
+    commits = json.loads(urlopen(req).read())
+    print('Retrieved', len(commits), 'commits')
+
+    # Load PR for each commit and print summary
+    for commit in commits:
+        sha = commit['sha']
+        author = commit['author']['login']
+
+        if exclude_dependabot and author == "dependabot[bot]":
+            skipped_dependabot += 1
+            continue
+
+        author_url = commit['author']['html_url']
+        msg_lines = commit['commit']['message'].split('\n')
+        msg = msg_lines[0]
+        req = Request(f"{commits_url}/{sha}/pulls")
+        req.add_header('accept', accept_header)
+        req.add_header('Authorization', f'token {token}')
+        pulls = json.loads(urlopen(req).read())
+        if len(pulls) > 1:
+            print(f"WARNING: More than one pull request for commit {sha}")
+
+        # Handle commits without pull requests.
+        if not pulls:
+            short_sha = sha[:7]
+            commit_url = commit['html_url']
+            print(f'* {msg} ([@{author}]({author_url}) in [{short_sha}]({commit_url}))')
+            continue
+
+        pull = pulls[0]
+        pull_id = pull['number']
+        pull_url = pull['html_url']
+        msg = msg.replace(f'(#{pull_id})', '').strip()
+        print(f'* {msg} ([@{author}]({author_url}) in [#{pull_id}]({pull_url}))')
+
+    if skipped_dependabot:
+        print(f"\n(Skipped {skipped_dependabot} dependabot commit{'' if skipped_dependabot == 1 else 's'})")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='List changes based on git log for release notes.')
+
+    parser.add_argument('--token-file', type=str, default="~/.github_token",
+                        help='The file containing your personal github token to access the github API. ' +
+                             '(default: ~/.github_token)')
+    parser.add_argument('--repo', type=str, default='jaeger',
+                        help='The repository name to fetch commit logs from. (default: jaeger)')
+    parser.add_argument('--exclude-dependabot', action='store_true',
+                        help='Excludes dependabot commits. (default: false)')
+    parser.add_argument('--num-commits', type=int,
+                        help='Print this number of commits from git log. ' +
+                             '(default: number of commits before the previous tag)')
+
+    args = parser.parse_args()
+    generate_token_url = "https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token"
+    generate_err_msg = (f"Please generate a token from this URL: {generate_token_url} and "
+                        f"place it in the token-file. The token only needs default permissions.")
```

Comment:
> The token only needs default permissions.

Not sure what "default permissions" mean - I would instead explicitly recommend `chmod 0600`

Reply:
```suggestion
                        f"place it in the token-file. Protect the file so only you can read it: chmod 0600 <file>.")
```

Reply:
Ah, I wanted to respect what was stated in the original [Makefile comment](https://github.com/jaegertracing/jaeger/blob/master/Makefile#L365). I thought it was referring to the privileges (scopes) given to the generated token in terms of API access. You can see them when clicking on this link: https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token

Perhaps it could be worded a little better.